xGame/client/src/Components/Match/MatchBoard.js

```diff
@@ -55,7 +61,12 @@ const MatchBoard = (props) => {
   ];
 
   const selectPiece = (selectedPieceData) => {
-    if (activeMatch.status === "INPROGRESS") {
+    if (
+      activeMatch.status === "INPROGRESS" &&
+      (activeColor === "white"
+        ? playerId === activeMatch.whitePlayerId
+        : playerId === activeMatch.blackPlayerId)
```

Comment:
This check should be when the makeMove service is about to be called, that way players can still see legal moves even if it isn't their turn

Reply:
Disagree. Their legal moves don't matter if its not their turn, since the opponents move will change your legal moves.

Reply:
Yeah, I don't think this would be a hugely crucial change